### PR TITLE
Added validation and testing for models

### PR DESCRIPTION
**Pull request by Htetaungkyaw71**
# This branch is for validation models and testing validate models

- Added the following validations ✅ 
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.  ✅

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+RSpec.describe User, type: :model do
+  subject { User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.') }
+  before { subject.save }
+
+  it 'name should be present' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+  it 'postCounter should be greater than and equal to zero' do
```
Comment (fobadara) [inline spec/models/user_spec.rb]: [OPTIONAL] I think you made a mistake in the description. You should have used `or` that is `greater than or equal`

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  end
+  it 'CommentCounter should be greater than and equal to zero' do
+    subject.comments_counter = -1
+    expect(subject).to_not be_valid
+  end
+  it 'CommentCounter should be integer' do
+    subject.comments_counter = 'hello'
+    expect(subject).to_not be_valid
+  end
+  it 'LikeCounter should be greater than and equal to zero' do
```
Comment (fobadara) [inline spec/models/post_spec.rb]: [OPTIONAL] I think you made a mistake in the description. You should have used or that is greater than or equal

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  it 'name should be present' do
+    subject.title = nil
+    expect(subject).to_not be_valid
+  end
+  it 'Title should be not exceed 250 characters.' do
+    subject.title = 'a' * 260
+    expect(subject).to_not be_valid
+  end
+  it 'CommentCounter should be greater than and equal to zero' do
```
Comment (fobadara) [inline spec/models/post_spec.rb]: [OPTIONAL] I think you made a mistake in the description. You should have used or that is greater than or equal

File: app/models/post.rb [hunk 4]
```diff
@@ -3,6 +3,11 @@ class Post < ApplicationRecord
   has_many :likes
   has_many :comments
 
+  validates :title, presence: true, length: { in: 0..250 }
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
+
   after_save :update_postcounter, :recent_five_comments
```
Comment (fobadara) [inline app/models/post.rb]: [REQUIRED] Kindly include tests for your custom methods. I can't see test for `:update_postcounter and :recent_five_comments`

File: app/models/user.rb [hunk 5]
```diff
@@ -2,6 +2,8 @@ class User < ApplicationRecord
   has_many :likes
   has_many :posts, foreign_key: 'author_id'
   has_many :comments
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
   after_save :recent_three_posts
```
Comment (fobadara) [inline app/models/user.rb:8]: [REQUIRED] Kindly include tests for your custom methods.

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @Htetaungkyaw71, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Linters are passing
✅ Good first attempt

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @Htetaungkyaw71, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**